### DISCUSSION TITLE
Line Arrow shapes not showing up in Shapes list

### DISCUSSION THREAD

**Discussion by grandoth** [ShapeCrawler/ShapeCrawler#203] [Q&A] [answered]
I have a very simple power point file that I am experimenting with. It contains a simple rectangle and two "Line Arrow" shapes (connector type). I can successfully load this Power Point file using 0.26.0 in a simple .NET Framework 4.8 WinForms project. However after accessing the first slide and enumerating over the .Shapes collection, there is only one item and it representing the rectangle. The two "Line Arrow" shapes are missing.

Is there a way to access the "Line" type shapes?

**Comment by AdamShakhabov** [comment]
No, Connector Shape (Line) is not supported yet.

**Comment by AdamShakhabov** [comment, marked as answer]
@grandoth, Connection shapes support has been added in the [**v0.27.0**](https://github.com/ShapeCrawler/ShapeCrawler/releases/tag/v0.27.0) version.

Feel free to ask if it doesn't work as expected.

**Reply by grandoth** [reply]
Thanks for working on this in such short order! I can access the main members of the IShape interface.

For my use case, more data, specifically connector path/points are important. I have not reviewed the [DocumentFormat.OpenXml.Drawing API much at all.

My end goal: To be able to parse shapes and connector shapes/paths and convert/import them to a simple CAD type drawing.

So far I have relied on PPT and saving as SVG. Then I use InkScape to convert that to DXF. However, instead of connector path/points, I get the rendered shape representation as one would expect. I can draw overlayed lines over the rendered elements by inspecting visually, but that requires a human in the loop.

To simplify the process, the path points/entities that make up the connector shape would be helpful:
- Singular or composite straight line / circular arc. Specifically properties of straight lines (endpoints) and circular arc (center, radius, end points and/or start/end angle). I'm not as interested in the more complex, connector shapes
- Snap points. This is if a connector shape is "snapped" to connector and/or shape including the reference/ID
- Some shape properties, specifically the head head type would be helpful to know if the arrow is (bi)directional. Line thickness/pattern is not as import.

I don't know how feasible this is and whether or not this would be on your roadmap. I appreciate any thoughts you have.

**Comment by AdamShakhabov** [comment]
Yes, I think it is suit with project roadmap, because one of the main goal is making possible to **extract (read)** any attribute of slide/shape.

In regard to feasibility, should be possible. Could you share some real usage slide (without sensitive data) that contains such lines? It would narrow lines' investigation down.

If I have some updates related to Connection shape type, I will share here...

**Reply by grandoth** [reply]
Here is an example with shapes connected with connector lines via snapping and anchor (link) points. PPT doesn't support every shape vertex being an anchor point. See the square for example. Its corners are not anchor points unless you customize it via Edit Points. Additionally connector lines can snap to each other, but are not anchored (linked). I think it is a good starting point for investigation. Thanks!
[PPT Example.pptx](https://github.com/ShapeCrawler/ShapeCrawler/files/8004146/PPT.Example.pptx)

**Comment by AdamShakhabov** [comment]
The current Line Arrow supported APIs can be checked [**here**](https://github.com/ShapeCrawler/ShapeCrawler?tab=readme-ov-file#line).
